src/meta/meta_backup_service.h

```diff
@@ -311,6 +312,51 @@ mock_private :
     dsn::task_tracker _tracker;
 };
 
+enum backup_status
+{
+    UNALIVE = 1,
+    ALIVE = 2,
+    COMPLETED = 3,
+    FAILED = 4
+};
+
+class backup_engine
+{
+public:
+    backup_engine(backup_service *service);
+    ~backup_engine();
+
+    error_code init_backups(const std::set<int32_t> &app_ids);
+    error_code set_block_service(const std::string &provider);
+    error_code write_backup_file(const std::string &file_name, const dsn::blob &write_buffer);
+    error_code backup_app_meta(const std::string &app_name, int32_t app_id);
+    void backup_app_partition(const std::string &app_name,
+                              const gpid &pid,
+                              const std::string &policy_name);
+    error_code write_backup_info();
+    void on_backup_reply(dsn::error_code err,
+                                 backup_response &&response,
+                                 gpid pid,
+                                 const rpc_address &primary);
+    error_code run();
+    bool is_backup_complete(std::string &hint_message);
+    int64_t get_current_backup_id() { return _cur_backup.backup_id; }
+
+private:
+    std::string get_policy_name() { return "fake_policy_" + std::to_string(_cur_backup.backup_id); }
+
+    backup_service *_backup_service;
+    dist::block_service::block_filesystem *_block_service;
+    std::string _provider_type;
+    dsn::task_tracker _tracker;
+
+    // lock _cur_backup and _backup_status.
+    dsn::zlock _lock;
+    backup_info _cur_backup;
+    // app_id --> partition_id, backup_status
+    std::map<int32_t, std::map<int32_t, backup_status>> _backup_status;
+};
```

Comment:
I prefer to move the lines into a new file, maybe backup_engine.h. It would be easier for readers to locate the class via file name.